precompile/contracts/ibc/client_test.go

```diff
@@ -0,0 +1,463 @@
+package ibc
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ava-labs/subnet-evm/core/state"
+	"github.com/ava-labs/subnet-evm/precompile/contract"
+	"github.com/ava-labs/subnet-evm/precompile/testutils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+
+	tmtypes "github.com/cometbft/cometbft/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	cosmostypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/std"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v7/modules/core/23-commitment/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+	ibctm "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
+	ibctesting "github.com/cosmos/ibc-go/v7/testing"
+	ibctestingmock "github.com/cosmos/ibc-go/v7/testing/mock"
+)
+
+const (
+	testChainID          = "gaiahub-0"
+	testChainIDRevision1 = "gaiahub-1"
+
+	testClientID1 = "07-tendermint-0"
+	testClientID2 = "07-tendermint-1"
+	testClientID3 = "07-tendermint-2"
+
+	height = 5
+
+	trustingPeriod time.Duration = time.Hour * 24 * 7 * 2
+	ubdPeriod      time.Duration = time.Hour * 24 * 7 * 3
+	maxClockDrift  time.Duration = time.Second * 10
+)
+
+var (
+	privVal                   = ibctestingmock.NewPV()
+	testClientHeight          = types.NewHeight(0, 5)
+	testClientHeightRevision1 = types.NewHeight(1, 5)
+
+	validator  *tmtypes.Validator
+	validators *tmtypes.ValidatorSet
+)
+
+func init() {
+	pubKey, _ := privVal.GetPubKey()
+	validator = tmtypes.NewValidator(pubKey, 1)
+	validators = tmtypes.NewValidatorSet([]*tmtypes.Validator{validator})
+}
+
+func TestCreateClient(t *testing.T) {
+	createClientOuput, err := PackCreateClientOutput(testClientID1)
+	require.NoError(t, err)
+
+	tests := map[string]testutils.PrecompileTest{
+		"success: 07-tendermint client type supported": {
+			Caller: common.Address{1},
+			InputFn: func(t testing.TB) []byte {
+				clientState, err := ibctm.
+					NewClientState(
+						testChainID,
+						ibctm.DefaultTrustLevel,
+						trustingPeriod,
+						ubdPeriod,
+						maxClockDrift,
+						testClientHeight,
+						commitmenttypes.GetSDKSpecs(),
+						ibctesting.UpgradePath,
+					).
+					Marshal()
+				require.NoError(t, err)
+
+				consensusState, err := ibctm.
+					NewConsensusState(
+						time.Now(),
+						commitmenttypes.NewMerkleRoot([]byte("hash")),
+						validators.Hash(),
+					).
+					Marshal()
+				require.NoError(t, err)
+
+				input, err := PackCreateClient(CreateClientInput{
+					ClientType:     exported.Tendermint,
+					ClientState:    clientState,
+					ConsensusState: consensusState,
+				})
+				require.NoError(t, err)
+				return input
+			},
+			SuppliedGas: CreateClientGasCost,
+			ReadOnly:    false,
+			ExpectedErr: "",
+			ExpectedRes: createClientOuput,
+		},
```

Comment:
@n0cte please add additional test when we pass non-tendermint light client type and get error client not supported